src/json.hpp

```diff
@@ -144,6 +337,41 @@ struct DecimalSeparator : std::numpunct<char>
 
 }
 
+// taken from ranges-v3
+// TODO add doc
+template <typename T>
+struct _static_const
```

Comment:
nitpick: the leading underscore _might_ be undefined behavior (I always forget the rules...). Worth checking.